### PR TITLE
Players: Always reset CDataCacheCore on close file

### PR DESCRIPTION
**Pull request by enen92**
## Description
Found while trying to extend datacachecore. As far as I remember this class was created as a cache directory for players to store information so that the GUI thread (info interface) could obtain the info without stalling the players. Setting a break point on `Reset()` I've found this is only called by the application when playing a new file. Means that state information is left in memory for no reason after a file is closed by the player.
Just call reset when all players Close the file.

## Motivation and context
Better usage of memory resources. In this case it doesn't really make much difference since it just resets the state back to the initial state and the structures sizes are the same. But if for some reason you want to store dynamic information in the cache (e.g. I want to store a pointer to the item being played) it'll be left in memory instead of being garbage collected. 

## How has this been tested?
Runtime tested on linux

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/RetroPlayer/RetroPlayer.cpp
Comment (enen92) [inline xbmc/cores/RetroPlayer/RetroPlayer.cpp:261]: @garbear any reason you don't simply call `m_processInfo->Reset()` in `ResetInfo()` ?

Comment (garbear) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp:261]: Probably because I wrote this in 2012 and Reset() was added in 2017. Can you change it?

Comment (garbear) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp:261]: Also guard this with `if (m_processInfo)`. CloseFile() is called from the destructor to ensure consistent destruction order in case the external application fails to call CloseFile() for some reason. So CloseFile() could be called twice. Or remove the call from the destructor and ensure the application can't reset the player core without calling CloseFile().

**Comment by enen92** [comment]
It'd be nice to get feedback here. This has low-risk, however I think it'd be great to at least have an approval (especially since it touches retroplayer which has active maintainers - ping @garbear ).
I could merge "unless someone opposes" but doesn't seem correct in this case.

**Review by ksooo — APPROVED** [review]
Looks sane.

File: xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp
Comment (garbear) [inline xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp:143]: It looks like `CDataCacheCore::Reset()` resets far less than this, do you think that's a problem?

Comment (enen92) [reply xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp:143]: I don't think it actually causes problems but it should reset all the members. I'll try to address this.

Comment (enen92) [reply xbmc/cores/RetroPlayer/process/RPProcessInfo.cpp:143]: Done

File: xbmc/cores/DataCacheCore.cpp
Comment (garbear) [inline xbmc/cores/DataCacheCore.cpp:56]: What about `m_hasAVInfoChanges`? Should that also be reset?

Comment (enen92) [reply xbmc/cores/DataCacheCore.cpp:56]: done, missed that one

**Pull request merged**
